app/models/auth_service_account.rb

```diff
@@ -0,0 +1,6 @@
+class AuthServiceAccount < Authentication
```

Comment:
A lot of this would be easier with just AuthUseridPassword, and I did get things working at one point just as a test by stuffing the json service_key into the password field and leaving `userid = '_'`.  Seemed a little hackey but it is definitely and option.